File: c/1/1.c
#include <stdio.h>
#include <stdlib.h> // atoi

int getfuel(int mass) {
	int fuel;
	fuel = (mass / 3) - 2; // typical div should handle rounding down

	return fuel;
}

int main(int argc, char **argv) {
	int fuel, mass;
	char *line;
	ssize_t linelen;
	size_t linecap;
	unsigned long long total;
	FILE *fp;

	fp = fopen("./input", "r");
	if (!fp) {
		fprintf(stderr, "Cannot open input file\n");
		return 1;
	}

	total = 0;
	for(linelen = 0, line = NULL; (linelen = getline(&line, &linecap, fp)) > 0; ) {
		mass = atoi(line);
		fuel = getfuel(mass);
		//printf("%d\n", fuel);
		total += fuel;

		linelen = 0;
		line = NULL;
	}

	printf("%llu\n", total);
	return 0;
}

File: go/1/1.go
package main

import (
	"fmt"
	"os"
	"bufio"
	"log"
	"strconv"
)

func getfuel(mass int) (fuel int) {
	fuel = (mass / 3) - 2
	return fuel
}

func main() {
	file, err := os.Open("./input")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	var total uint64
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		mass, _ := strconv.Atoi(scanner.Text())
		fuel := getfuel(mass)
		total += uint64(fuel)
	}

	fmt.Println(total)
}
